engine/beef_tx_sorting.go

```diff
@@ -20,14 +20,14 @@ func kahnTopologicalSortTransactions(transactions []*bt.Tx) []*bt.Tx {
 	return result
 }
 
-func prepareSortStructures(dag []*bt.Tx) (txByID map[string]*bt.Tx, incomingEdgesMap map[string]int, zeroIncomingEdgeQueue []string) {
+func prepareSortStructures(dag []*trx.Transaction) (txByID map[string]*trx.Transaction, incomingEdgesMap map[string]int, zeroIncomingEdgeQueue []string) {
 	dagLen := len(dag)
-	txByID = make(map[string]*bt.Tx, dagLen)
+	txByID = make(map[string]*trx.Transaction, dagLen)
 	incomingEdgesMap = make(map[string]int, dagLen)
 
 	for _, tx := range dag {
-		txByID[tx.TxID()] = tx // TODO: perf -> In bt, the TxID is calculated every time we try to get it, which means we hash the tx bytes twice each time. It's expensive operation - try to avoid calculation each time
-		incomingEdgesMap[tx.TxID()] = 0
+		txByID[tx.TxID().String()] = tx // TODO: perf -> In GO-SDK, the TxID is calculated every time we try to get it, which means we hash the tx bytes twice each time. It's expensive operation - try to avoid calculation each time
+		incomingEdgesMap[tx.TxID().String()] = 0
```

Comment:
So maybe, let's apply the suggestion from the comment 🙈 
```suggestion
                txID := tx.TxID().String()
		txByID[txID] = tx 
		incomingEdgesMap[txID] = 0
```